ui/ui-components/pages/model/[modelId]/source/[sourceId]/edit.tsx

```diff
@@ -28,22 +29,24 @@ import PrimaryKeyBadge from "../../../../../components/badges/PrimaryKeyBadge";
 import { useApi } from "../../../../../contexts/api";
 import { generateClient } from "../../../../../client/client";
 import { withServerErrorHandler } from "../../../../../utils/withServerErrorHandler";
+import { SourcePreviewMethodResponseRow } from "@grouparoo/core/src/classes/plugin";
 
-interface FormData {
+export interface FormData {
   mapping?: {
     sourceColumn: string;
     propertyKey: string;
   };
-  source: Pick<Models.SourceType, "name" | "options">;
+  source: Pick<Actions.SourceView["source"], "name" | "options">;
 }
 
 interface Props {
   environmentVariableOptions: Actions.AppOptions["environmentVariableOptions"];
-  properties: Models.PropertyType[];
+  properties: Actions.PropertiesList["properties"];
   propertyExamples: Record<string, string[]>;
   scheduleCount: number;
-  source: Models.SourceType;
+  source: Actions.SourceView["source"];
```

Comment:
Not a big fan of pulling types out of function calls. The confusing part is that we have two ways to type what a source is, but there should only be one.

Reply:
Well this is the direct response from the server so it's accurate to the call. 🤷 
There's issues reaching into core for the correct type so we make due.